oadr/oadr/manager/eventmanager/EventManager.cpp

```diff
@@ -219,23 +219,37 @@ void EventManager::handleExistingEvent(const string &eventID, const oadr2b::oadr
 
 	if (event->eiEvent().eventDescriptor().eventStatus() == EventStatusEnumeratedType::cancelled)
 	{
-		oadr2b::ei::OptTypeType::value optType = oadr2b::ei::OptTypeType::optIn;
+		if (oadr2b::oadr::ResponseRequiredType::value::always == event->oadrResponseRequired())
+		{
+			oadr2b::ei::OptTypeType::value optType = oadr2b::ei::OptTypeType::optIn;
 
-		m_service->OnEventCancel(eventID, event, optType);
+			m_service->OnEventCancel(eventID, event, optType);
 
-		Oadr2bHelper::appendEventResponse(eventResponses, "200", "OK", eventID,
-				event->eiEvent().eventDescriptor().modificationNumber(), optType, requestID);
+			Oadr2bHelper::appendEventResponse(eventResponses, "200", "OK", eventID,
+					event->eiEvent().eventDescriptor().modificationNumber(), optType, requestID);
+		}
+		else // oadr2b::oadr::ResponseRequiredType::value::never
+		{
+			m_service->OnEventCancel(eventID, event);
```

Comment:
The wording of the conformance rule 12:
VEN, EiEvent Service, oadrCreatedEvent Payload

The VEN MUST respond with an oadrCreatedEvent to an event in oadrDistributeEvent based upon the value in each event’s oadrResponseRequired element as follows:

Always – The VEN MUST respond to the event with an oadrCreatedEvent eventResponse. This includes unchanged, new, changed, and canceled events.

Never – The VEN MUST NOT respond to the event with a oadrCreatedEvent eventResponse

Note that oadrCreatedEvent event responses SHOULD be returned in one message, but MAY be returned in separate messages.